### DISCUSSION TITLE
Comntributor/contribution exclusion issue

### DISCUSSION THREAD

**Discussion by fbiville** [jreleaser/jreleaser#955] [Q&A] [answered]
Hello, 

There must be something I misconfigured with the JReleaser Maven plugin for https://github.com/liquibase/liquibase-neo4j, because I still see contributors and contributions included in the generated change log even though I want to exclude then.

# The Setup

Here is the generated change log (note: [the actual one](https://github.com/liquibase/liquibase-neo4j/releases/tag/liquibase-neo4j-4.16.1.1) has been manually fixed up) for version 1.16.1.1:

```markdown
## Changelog

- 12c3da7 Set next release version (Florent Biville)
- 41fbfa1 Fix contribution exclusion from change log (Florent Biville)
- d8cfdef Support <loadData /> (Andrea Santurbano)
- e60822b Version Bumped to Snapshot for Development (liquibot)
- faf38ae Version Bumped to 4.16.1 (liquibot)
- 745ed16 Bump liquibase-core from 4.16.0 to 4.16.1 (dependabot[bot])
- c60e0b0 Bump maven-shade-plugin from 3.3.0 to 3.4.0 (dependabot[bot])
- 6eb9ed3 Bump slf4j-nop from 1.7.36 to 2.0.1 (dependabot[bot])
- c6ffe90 Add missing copyright (Florent Biville)
- 1a4dd9f Version Bumped to Snapshot for Development (liquibot)
- 62bcf1a Version Bumped to 4.16.0 (liquibot)
- f3d561e Bump liquibase-core from 4.15.0 to 4.16.0 (dependabot[bot])


## Contributors
We'd like to thank the following people for their contributions:
dependabot[bot], Andrea Santurbano, Florent Biville, liquibot
```

The Maven plugin is [configured](https://github.com/liquibase/liquibase-neo4j/blob/5b6148e1e8e8ed47789a73815ba64edf1c8f3442/pom.xml#L340-L376) as follows:

```xml
<release>
                                    <github>
                                        <!-- [...] -->
                                        <changelog>
                                            <preset>conventional-commits</preset>
                                            <formatted>ALWAYS</formatted>
                                            <hide>
                                                <contributors>
                                                    <contributor>GitHub</contributor>
                                                    <contributor>.*bot</contributor>
                                                </contributors>
                                            </hide>
                                        </changelog>
                                    </github>
                                </release>
```

The plugin is followed when the `published` event is triggered in GitHub Actions (full workflow [here](https://github.com/liquibase/liquibase-neo4j/blob/5b6148e1e8e8ed47789a73815ba64edf1c8f3442/.github/workflows/release-published.yml)):

```yaml
        - name: Populate release change log
          run: mvn --quiet --batch-mode jreleaser:release -P release
          env:
            JRELEASER_GITHUB_TOKEN: ${{ secrets.JRELEASER_GITHUB_TOKEN }}
```

# Exclusion issue

I expected both `liquibot` and `dependabot` to be excluded as well as their contributions, but that's not the case, as you can see above.

What am I missing here?

**Comment by aalmiray** [comment, marked as answer]
I think it's enough to say `bot` or alternatively `*bot`. You can run `mvn jreleaser:changelog` and inspect `target/jreleaser/release/CHANGELOG.md` to see if these settings produced the expected result or not. If the regex did not work then that's likely a bug.

**Reply by fbiville** [reply]
Thanks I'll give it a try now.
I realize what I want is a bit more complicated.
I want both `liquibot` and `dependabot` to be excluded from contribut**or**s.
But I only want `liquibot` contribut**ion**s to be excluded.
Is there a way to do this?

**Reply by fbiville** [reply]
I tried both `bot` and `*bot`. Only `bot` works. In the second case, both `liquibot` and `dependabot[bot]` show up.

**Reply by aalmiray** [reply]
Well, kind of. If the liquibot contributions follow a particular pattern in their title/body then yes, you can match those with a label, categorized them and hide the given category. If that's not the case then it can't be done, yet. But it will be possible soon, once we release the extensions API you'll be able to hook in custom changelog processors.

**Reply by aalmiray** [reply]
Alright, this means the regex eval is buggy. Could you file a ticket? Thanks 😄

**Reply by fbiville** [reply]
Issue filed: https://github.com/jreleaser/jreleaser/issues/957
Lucky for me, `liquibot` messages are always starting with `Version Bumped to`

**Reply by fbiville** [reply]
https://github.com/liquibase/liquibase-neo4j/releases/tag/liquibase-neo4j-4.16.1.2 🤩
